TetrisDotnet/Code/Events/EventSystem.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+
+namespace TetrisDotnet.Code.Events
+{
+    public delegate void OnEvent(EventData.EventData eventData);
+
+    public delegate void OnTrigger();
```

Comment:
Not quite. One of them allows you to package data and send it through. OnTrigger is oh-so-much simpler, so for cases where you only need to know "when" something happens, and not "what", it's easier to go the OnTrigger route.

Do you have a more generic solution that would allow us to still allow both via the same delegate?

Reply:
You can just usue OnEvent and ignore the event data, basically OnTrigger